### PR TITLE
Organized VC Contents and Added Documentation

### PR DESCRIPTION
**Pull request by rs929**
## Overview

Organized and Annotated All VCs
Added Documentation to functions

### CODE REVIEW

File: Scoop/Controller/Home/HomeViewController.swift [hunk 1]
```diff
@@ -10,20 +10,23 @@ import UIKit
...
-    // MARK: Empty State Views
+    // MARK: - Empty State Views
+    
     private let noTripsLabel = UILabel()
     private let searchButton = UILabel()
     private let secondLabel = UILabel()
     private let widePostRideButton = UIButton()
     
-    // MARK: Identifers
+    // MARK: - Identifers
```
Comment (vinnie4k) [inline Scoop/Controller/Home/HomeViewController.swift]: Identifiers

File: Scoop/Controller/Home/NotificationsViewController.swift [hunk 2]
```diff
@@ -81,12 +90,12 @@ class NotificationsViewController: UIViewController {
     
 }
 
-// MARK: UITableViewDelegate
+// MARK: - UITableViewDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Home/NotificationsViewController.swift:93]: Keep it consistent and add a new line below.

File: Scoop/Controller/Home/NotificationsViewController.swift [hunk 3]
```diff
@@ -81,12 +90,12 @@ class NotificationsViewController: UIViewController {
...
 extension NotificationsViewController: UITableViewDelegate {
     // Will be used later for updating what the cell looks like when a user hits accept/decline
 }
 
-// MARK: UITableViewDataSource
+// MARK: - UITableViewDataSource
```
Comment (vinnie4k) [inline Scoop/Controller/Home/NotificationsViewController.swift:99]: Keep it consistent and add a new line below.

File: Scoop/Controller/LocationViewController.swift [hunk 4]
```diff
@@ -57,11 +64,14 @@ class LocationViewController: UIViewController {
...
     }
     
+    // MARK: - Helper Functions
+    
     func filterText(searchText: String) {
         tableDataSource.sourceTextHasChanged(searchText)
     }
 }
 
+// MARK: - GMSAutocompleteTableDataSourceDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/LocationViewController.swift:74]: Keep it consistent and add a new line below.

File: Scoop/Controller/MatchesViewController.swift [hunk 5]
```diff
@@ -314,6 +325,7 @@ class MatchesViewController: UIViewController {
 
 }
 
+// MARK: - UITableViewDataSource
```
Comment (vinnie4k) [inline Scoop/Controller/MatchesViewController.swift:328]: Keep it consistent and add a new line below.

File: Scoop/Controller/MatchesViewController.swift [hunk 6]
```diff
@@ -328,6 +340,7 @@ extension MatchesViewController: UITableViewDataSource {
     
 }
 
+// MARK: - UITableViewDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/MatchesViewController.swift:344]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/FavoritesViewController.swift [hunk 7]
```diff
@@ -151,6 +157,7 @@ class FavoritesViewController: OnboardingViewController {
     
 }
 
+// MARK: - UITextFieldDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift:160]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/OnboardingContainerViewController.swift [hunk 8]
```diff
@@ -51,6 +54,7 @@ class OnboardingContainerViewController: ViewController {
     
 }
 
+// MARK: - AnimationDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/OnboardingContainerViewController.swift:58]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/OnboardingContainerViewController.swift [hunk 9]
```diff
@@ -11,12 +11,15 @@ class OnboardingContainerViewController: ViewController {
     
     let screenMultiplier = 335.0/375.0
     
+    // MARK: - Views
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/OnboardingContainerViewController.swift:14]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/OnboardingViewController.swift [hunk 10]
```diff
@@ -15,16 +15,21 @@ protocol OnboardingDelegate: AnyObject {
 class OnboardingViewController: UIViewController {
     
     weak var delegate: OnboardingDelegate?
+    
+    // MARK: - Views
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift:19]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/OnboardingViewController.swift [hunk 11]
```diff
@@ -15,16 +15,21 @@ protocol OnboardingDelegate: AnyObject {
...
     var nextAction: UIAction?
     let backButton = UIButton()
     let nextOnboardingButton = UIButton()
-
+    
+    // MARK: - Lifecycle Functions
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift:25]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/PreferencesViewController.swift [hunk 12]
```diff
@@ -8,11 +8,14 @@
 import UIKit
 
 class PreferencesViewController: OnboardingViewController {
-
+    
+    // MARK: - Views
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/PreferencesViewController.swift:12]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/PreferredContactViewController.swift [hunk 13]
```diff
@@ -183,6 +191,7 @@ class PreferredContactViewController: OnboardingViewController {
     
 }
 
+// MARK: - UITextFieldDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/PreferredContactViewController.swift:194]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/ProfilePictureViewController.swift [hunk 14]
```diff
@@ -197,12 +158,61 @@ class ProfilePictureViewController: OnboardingViewController {
...
+        presentErrorAlert(title: "Unable to Create Account", message: "There was an error while creating your account. Please try again at another time.")
+    }
+    
     @objc private func uploadProfilePicture(_ sender: UITapGestureRecognizer) {
         present(imagePicker, animated: true)
     }
 }
 
-// MARK: UIImagePickerControllerDelegate
+// MARK: - UIImagePickerControllerDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/ProfilePictureViewController.swift:215]: Keep it consistent and add a new line below.

File: Scoop/Controller/Onboarding/VerifyPhoneNumberViewController.swift [hunk 15]
```diff
@@ -162,27 +191,9 @@ class VerifyPhoneNumberViewController: UIViewController {
...
-        codeTextField.snp.makeConstraints { make in
-            make.width.equalTo(120)
-            make.height.equalTo(56)
-            make.center.equalToSuperview()
-        }
-    }
-    
 }
 
+// MARK: - UITextFieldDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/VerifyPhoneNumberViewController.swift:196]: Keep it consistent and add a new line below.

File: Scoop/Controller/Ride Posting/InitialPostRideViewController.swift [hunk 16]
```diff
@@ -279,6 +287,7 @@ class InitialPostRideViewController: PostRideViewController {
     
 }
 
+// MARK: - SearchInitialViewControllerDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Ride Posting/InitialPostRideViewController.swift:290]: Keep it consistent and add a new line below.

File: Scoop/Controller/Ride Posting/PostRideContainerViewController.swift [hunk 17]
```diff
@@ -94,6 +100,7 @@ class PostRideContainerViewController: ViewController {
     
 }
 
+// MARK: - AnimationDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Ride Posting/PostRideContainerViewController.swift:103]: Keep it consistent and add a new line below.

File: Scoop/Controller/Ride Posting/PostRidePageViewController.swift [hunk 18]
```diff
@@ -96,6 +104,7 @@ class PostRidePageViewController: UIPageViewController, UIPageViewControllerDele
     
 }
 
+// MARK: - UIPageViewControllerDataSource
```
Comment (vinnie4k) [inline Scoop/Controller/Ride Posting/PostRidePageViewController.swift:107]: Keep it consistent and add a new line below.

File: Scoop/Controller/Ride Posting/PostRidePageViewController.swift [hunk 19]
```diff
@@ -126,6 +135,7 @@ extension PostRidePageViewController: UIPageViewControllerDataSource {
     
 }
 
+// MARK: - OnboardingDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Ride Posting/PostRidePageViewController.swift:139]: Keep it consistent and add a new line below.

File: Scoop/Controller/Ride Posting/PostRideSummaryViewController.swift [hunk 20]
```diff
@@ -400,6 +407,7 @@ class PostRideSummaryViewController: PostRideViewController {
     }
 }
 
+// MARK: - PostRideDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Ride Posting/PostRideSummaryViewController.swift:410]: Keep it consistent and add a new line below.

File: Scoop/Controller/Ride Posting/PostRideTripDetailsViewController.swift [hunk 21]
```diff
@@ -344,6 +336,24 @@ class PostRideTripDetailsViewController: PostRideViewController {
         }
     }
     
+    // MARK: - Helper Functions
+    
+    private func getTravelDate() -> Date? {
+        let dateComponents = Calendar.current.dateComponents([.year, .month, .day], from: datePicker.date)
+
```
Comment (vinnie4k) [inline Scoop/Controller/Ride Posting/PostRideTripDetailsViewController.swift:343]: Remove this line

File: Scoop/Controller/Ride Posting/PostRideTripDetailsViewController.swift [hunk 22]
```diff
@@ -358,6 +368,7 @@ class PostRideTripDetailsViewController: PostRideViewController {
     
 }
 
+// MARK: - UITextFieldDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Ride Posting/PostRideTripDetailsViewController.swift:371]: Keep it consistent and add a new line below.

File: Scoop/Controller/Ride Posting/PostRideTripDetailsViewController.swift [hunk 23]
```diff
@@ -410,6 +421,7 @@ extension PostRideTripDetailsViewController: UITextFieldDelegate {
     
 }
 
+// MARK: - UITextViewDelegate
```
Comment (vinnie4k) [inline Scoop/Controller/Ride Posting/PostRideTripDetailsViewController.swift:425]: Keep it consistent and add a new line below.

File: Scoop/SearchRidesViewController.swift [hunk 24]
```diff
@@ -22,11 +23,12 @@ class SearchRidesViewController: UIViewController {
     private let findTripsButton = UIButton()
     private let stackView = UIStackView()
     
-    // MARK: Data
+    // MARK: - Data
```
Comment (vinnie4k) [inline Scoop/SearchRidesViewController.swift:26]: Keep it consistent and add a new line below.

File: Scoop/SearchRidesViewController.swift [hunk 25]
```diff
@@ -22,11 +23,12 @@ class SearchRidesViewController: UIViewController {
...
     private var arrivalPlace: GMSPlace?
     private var departurePlace: GMSPlace?
     private var tripDate: String = ""
     
+    // MARK: - Lifecycle Functions
```
Comment (vinnie4k) [inline Scoop/SearchRidesViewController.swift:32]: Keep it consistent and add a new line below.

**Review by vinnie4k — CHANGES_REQUESTED** [review]
Looks good! Just make sure you keep things consistent (i.e. line below for every `// MARK`). I know you just moved everything around, but next time document the files that you moved in the PR. Once you address these comments (make sure to mark them as resolved), request me again.

**Review by vinnie4k — APPROVED** [review]
Good to go! Next time add more information about what you moved around in the PR for documentation purposes, and also mark the comments as "Resolved" when you resolve them.

File: Scoop/Controller/Onboarding/VerifyPhoneNumberViewController.swift [hunk 26]
```diff
@@ -162,27 +191,10 @@ class VerifyPhoneNumberViewController: UIViewController {
...
+
 extension VerifyPhoneNumberViewController: UITextFieldDelegate {
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
```
Comment (tiffany-pann) [inline Scoop/Controller/Onboarding/VerifyPhoneNumberViewController.swift:199]: Line break here

**Review by tiffany-pann — APPROVED** [review]
LGTM! Fun PR to read 🚀

**Pull request merged**